concept/type/ThingTypeImpl.java

```diff
@@ -28,275 +28,278 @@
+    public final void unsetPlays(TypeDBTransaction transaction, RoleType roleType) {
+        thing_type_unset_plays(((ConceptManagerImpl) transaction.concepts()).transaction, nativeObject, ((RoleTypeImpl) roleType).nativeObject);
+    }
+
+    @Override
+    public final String getSyntax(TypeDBTransaction transaction) {
+        return thing_type_get_syntax(((ConceptManagerImpl) transaction.concepts()).transaction, nativeObject);
+    }
+
+    public static class Root extends ThingTypeImpl {
+        public Root(com.vaticle.typedb.client.jni.Concept concept) {
+            super(concept);
         }
 
         @Override
-        public AttributeTypeImpl getOwnsOverridden(AttributeType attributeType) {
-            ConceptProto.ThingType.GetOwnsOverridden.Res res = execute(
-                    getOwnsOverriddenReq(getLabel(), protoThingType(attributeType))
-            ).getThingTypeGetOwnsOverriddenRes();
-            switch (res.getResCase()) {
-                case ATTRIBUTE_TYPE:
-                    return AttributeTypeImpl.of(res.getAttributeType());
-                default:
-                case RES_NOT_SET:
-                    return null;
-            }
+        public final Label getLabel() {
+            return Label.of(TypeQLToken.Type.THING.toString());
         }
 
         @Override
-        public final void unsetPlays(RoleType roleType) {
-            execute(unsetPlaysReq(getLabel(), protoRoleType(roleType)));
+        public ThingTypeImpl getSupertype(TypeDBTransaction transaction) {
+            return null;
         }
 
         @Override
-        public final void unsetOwns(AttributeType attributeType) {
-            execute(unsetOwnsReq(getLabel(), protoThingType(attributeType)));
+        public Stream<? extends ThingTypeImpl> getSupertypes(TypeDBTransaction transaction) {
+            return null;
         }
 
         @Override
-        public ThingTypeImpl.Remote asRemote(TypeDBTransaction transaction) {
-            return new ThingTypeImpl.Remote(transaction, getLabel(), isRoot(), isAbstract());
+        public Stream<? extends ThingTypeImpl> getSubtypes(TypeDBTransaction transaction) {
+            return null;
         }
 
         @Override
-        public final ThingTypeImpl.Remote asThingType() {
-            return this;
+        public Stream<? extends ThingTypeImpl> getSubtypesExplicit(TypeDBTransaction transaction) {
+            return null;
         }
 
         @Override
-        public final boolean isDeleted() {
-            return transactionExt.concepts().getThingType(getLabel().name()) == null;
+        public Stream<? extends ThingImpl> getInstances(TypeDBTransaction transaction) {
+            return null;
         }
 
         @Override
-        public final String getSyntax() {
-            return execute(getSyntaxReq(getLabel())).getThingTypeGetSyntaxRes().getSyntax();
+        public Stream<? extends ThingImpl> getInstancesExplicit(TypeDBTransaction transaction) {
+            return null;
```

Comment:
This is strange, we didn't have a `Root` type before at all right? First of all `instances()` should still return things, not null, and secondly, why do we have a new structure we didn't have before?